src/tools/debugger/dosdebug.c

```diff
@@ -506,21 +508,27 @@ int main (int argc, char **argv)
         rl_callback_read_char();
 #else
         int num;
-        char *p = malloc(MHP_BUFFERSIZE), *q;
+        char buf[MHP_BUFFERSIZE], *p;
 
-        assert(p != NULL);
-
-        num = read(fdconin, p, MHP_BUFFERSIZE - 1);
-        if (num < 0) {
-          free(p);
+        num = read(fdconin, buf, sizeof(buf));
+        if (num < 0)
           break;
+
+        buf[MHP_BUFFERSIZE - 1] = '\0';
```

Comment:
Didn't you mean to write `buf[num] = '\0';`?

Reply:
Yes that would be better, becoming a little addled here.